packages/bento-design-system/src/FileUploaderField/FileUploaderField.tsx

```diff
@@ -141,10 +140,11 @@ export function FileUploaderField<E extends string>({
   maxFileSize,
   allowedFileTypes,
   height: height_,
+  isUploading,
 }: Props<E>) {
   const config = useBentoConfig().fileUploaderField;
   const height = height_ ?? config.defaultHeight;
-  const [uploading, setUploading] = useState<boolean>(false);
+  const [uploading, setUploading] = useState<boolean>(isUploading ?? false);
```

Comment:
I'm not sure this is making the component fully Controlled. In this way, the `isUploading` prop will be used only to initialize the state, but a change of the prop from outside will not have effect on the internal state.

If you want to change the state on prop change, you need to manually handle it with a `useEffect`:

```typescript
useEffect(() => {
   setUploading(isUploading);
}, [isUploading])
```

Reply:
Should be correct now:

https://user-images.githubusercontent.com/26444095/221237774-d70c0a72-b520-4f26-a482-68c35cb469de.mov